.github/.licenserc.yaml

```diff
@@ -0,0 +1,67 @@
+#===============================================================================
+# Copyright 2025 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+header:
+  license:
+    spdx-id: Apache-2.0
+    pattern: |
+      (Copyright \d{4} Intel Corporation|Copyright \d{4}(?:[,\s-]+\d{4})* (Arm Ltd\. and affiliates|Codeplay Software Limited|FUJITSU LIMITED|KNS Group LLC \(YADRO\)|IBM Corporation|Alanna Tempest|YANDEX LLC))
+      Licensed under the Apache License, Version 2\.0 \(the "License"\);
+      you may not use this file except in compliance with the License\.
+      You may obtain a copy of the License at
+
+          http:\/\/www\.apache\.org\/licenses\/LICENSE-2\.0
+
+      Unless required by applicable law or agreed to in writing, software
+      distributed under the License is distributed on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied\.
+      See the License for the specific language governing permissions and
+      limitations under the License\.
+
+  paths:
+    - '**'
+
+  paths-ignore:
+# all documentation-like files
+  - '**doc/**'
+  -  '**/*.md'
+
+# files in the root (which are typically are doc-like files)
+  - '*'
+
+# Github automation
+  - '**/*.github/**'
+  - '**/*.git*'
+
+# Clang tools configs
+  - '**/*.clang-*'
+
+# cmake
+  - 'cmake/ACL.cmake'
+  - 'cmake/FindACL.cmake'
+  - 'cmake/FindBLAS.cmake'
+  - 'cmake/FindOpenCL.cmake'
+  
+# benchdnn inputs
+  - 'tests/benchdnn/inputs/**'
+
+# Third party components
+  - 'third_party/**'
+
+# visual-studio files
+  - '**/*.vcxproj*'
+
+  license-location-threshold: 111  # specifies the index threshold where the license header can be located.
```

Comment:
Comment for this option is not particularly helpful. Assuming that it indicates the maximum number of lines from the top of the file to check something like `25` should be enough.